packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -18,6 +18,61 @@ export type Options = [Config];
 
 export type MessageIds = 'unbound';
 
+const nativelyBoundMembers = [
+  ...[
+    Promise,
+    //    BigInt, // todo: enable when we drop node@8
+    Number,
+    Object,
+    String,
+    RegExp,
+    Symbol,
+    Array,
+    Proxy,
+    Date,
+  ].map(x => [x.name, x]),
+  ['Infinity', Infinity],
+  ['Atomics', Atomics],
+  ['Reflect', Reflect],
+  ['console', console],
+  ['Math', Math],
+  ['JSON', JSON],
+  ['Intl', Intl],
+]
```

Comment:
you could probably make this using `global` instead:

```ts
const nativelyBoundMembers = [
  'Promise',
  'Number',
  ...
]
  .map(namespace => {
    const object = global[namespace];
    return Object.get.........
  })
  .reduce((arr, names) => arr.concat(names), []);
```

Reply:
This won't work because `Global` doesn't have an index type :/

Reply:
Ahh yeah, that's because TS just widens it to `string[]`.
An `as const` assertion will preserve the string values.

Reply:
Na I tried that - it still complains, as they resolve to their constructor name :)

```
([
  Number
] as const).map(n => global[n]);
// Error:(21, 29) TS2538: Type 'NumberConstructor' cannot be used as an index type.
```

Reply:
Sorry I'm working on both `@typescript-eslint` & `jest`; the monorepos are starting to blur together :joy:

The issue with that is that `global` doesn't hold everything in it's namespace, such as `Proxy` (according to TS at least), and also annoyingly for some reason `Math` doesn't have a name :/

Reply:
huh. that's just a bug in `@types/node`, as they definitely exist on `global`. That's how nodejs works - global is node's window.

It might be because `Proxy` is ES2015, and `Atomics` is ES2017, so you need libs for them to exist.

Feel free to create `packages/eslint-plugin/typings/node.d.ts`:

```ts
// augment nodejs global with ES2015+ things
declare namespace NodeJS {
  interface Global {
    Atomics: typeof Atomics;
    Proxy: typeof Proxy;
    Reflect: typeof Reflect;
  }
}
```

Then this works fine:
```ts
([
  'Promise',
  'Number',
  'Object',
  'String', // eslint-disable-line @typescript-eslint/internal/prefer-ast-types-enum
  'RegExp',
  'Symbol',
  'Array',
  'Proxy',
  'Date',
  'Infinity',
  'Atomics',
  'Reflect',
  'console',
  'Math',
  'JSON',
  'Intl',
] as const).map(name => global[name])
```